src/familiar.ts

```diff
@@ -55,177 +63,326 @@ export function meatFamiliar(): Familiar {
-    !get("_meatifyMatterUsed") &&
-    myInebriety() <= inebrietyLimit()
-  ) {
-    const experienceNeeded = 400 - (globalOptions.ascending ? gooseExp : 25);
-    const meatFromCast = 15 ** 4;
-    const estimatedExperience = 12;
-    familiarValue.push([
-      $familiar`Grey Goose`,
-      meatFromCast / (experienceNeeded / estimatedExperience),
-    ]);
+export function freeFightFamiliarData(canMeatify = false): GeneralFamiliar {
+  if (canMeatify && timeToMeatify()) {
+    return {
+      familiar: $familiar`Grey Goose`,
+      expectedValue: (familiarWeight($familiar`Grey Goose`) - 5) ** 4,
+      leprechaunMultiplier: 0,
+    };
   }
-  for (const familiarName of Object.keys(rotatingFamiliars)) {
-    const familiar: Familiar = Familiar.get(familiarName);
-    if (have(familiar)) {
-      const { expected, drop, pref } = rotatingFamiliars[familiarName];
-      const dropsAlready = get(pref);
-      if (dropsAlready >= expected.length) continue;
-      const value = garboValue(drop) / expected[dropsAlready];
-      familiarValue.push([familiar, value]);
+
+  const familiars = [
+    ...standardFamiliars(),
+    ...filterNull(experienceFamiliars.map(valueExperienceFamiliar)),
+    ...filterNull(rotatingFamiliars.map(valueStandardDropFamiliar)),
+  ];
+
+  const compareFams = (a: GeneralFamiliar, b: GeneralFamiliar) => {
+    if (a.expectedValue === b.expectedValue) {
+      return findLeprechaunMultiplier(a.familiar) > findLeprechaunMultiplier(b.familiar);
     }
-  }
+    return a.expectedValue > b.expectedValue;
+  };
 
-  if (have($familiar`Stocking Mimic`)) {
-    const mimicWeight = myFamiliarWeight($familiar`Stocking Mimic`);
-    const actionPercentage = 1 / 3 + (haveEffect($effect`Jingle Jangle Jingle`) ? 0.1 : 0);
-    const mimicValue = mimicDropValue() + ((mimicWeight * actionPercentage * 1) / 4) * 10 * 4 * 1.2;
-    familiarValue.push([$familiar`Stocking Mimic`, mimicValue]);
-  }
+  return familiars.reduce((a, b) => (compareFams(a, b) ? a : b));
+}
 
-  if (have($familiar`Obtuse Angel`)) {
-    familiarValue.push([$familiar`Obtuse Angel`, 0.02 * garboValue($item`time's arrow`)]);
-  }
+export function freeFightFamiliar(canMeatify = false): Familiar {
+  return freeFightFamiliarData(canMeatify).familiar;
+}
 
-  if (have($familiar`Robortender`)) familiarValue.push([$familiar`Robortender`, 200]);
+type MarginalFamiliar = GeneralFamiliar & { outfitValue: number };
+const outfitSlots = $slots`hat, back, shirt, weapon, off-hand, pants, acc1, acc2, acc3, familiar`;
+const cachedOutfits = new Map<number, { weight: number; meat: number; bonus: number }>();
 
-  for (const familiar of $familiars`Hobo Monkey, Cat Burglar, Urchin Urchin, Leprechaun`) {
-    if (have(familiar)) familiarValue.push([familiar, 1]);
-  }
+function cacheOutfit(
+  mult: number,
+  familiar: Familiar
+): { weight: number; meat: number; bonus: number } {
+  useFamiliar(familiar);
+  meatOutfit(false, undefined, false);
```

Comment:
I never did dive deep into the issue, but there were several instances where garbo decided the kramco was optimal for one outfit but not the other, resulting in a terribly wrong comparison between leps and non-leps.

In the ideal case, the `Requirements` here should enforce that the maximized result is exactly our normal barf outfit for that familiar. `undefined` doesn't seem to cut it, especially in edge cases relating to free fights.